Tools/ros2/ros2.repos

```diff
@@ -0,0 +1,9 @@
+repositories:
+  ardupilot:
+    type: git
+    url: https://github.com/srmainwaring/ardupilot.git
```

Comment:
Will this be migrated before PR merges?

Reply:
Yes. Although it will mean it can't be tested until the PR is merged. There's a bootstrap problem. We need the PR in to retrieve the cmake enabled build from master. Was thinking that for this and similar we might immediately follow this PR with a patch to set repo versions correct? Or I could just change it and you trust me it works.

Reply:
I'm fine with a patch